crates/router/src/core/routing/helpers.rs

```diff
@@ -216,7 +217,7 @@ pub async fn update_merchant_active_algorithm_ref(
         default_profile: None,
         payment_link_config: None,
     };
-
+    let config_key = cache::CacheKind::Config(key.into());
```

Comment:
You might want to create separate `CacheKind` variants for decision manager and surcharge, as from what I understand `CacheKind::Config` will cause an invalidation of `CONFIG_CACHE` only, not `DECISION_MANAGER_CACHE` and `SURCHARGE_CACHE`. Check `pub_sub.rs` for the invalidation logic.